src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/AgreementsApiServiceImpl.scala

```diff
@@ -490,4 +491,38 @@ final case class AgreementsApiServiceImpl(
     }
   }
 
+  override def getAgreementProducers(q: Option[String], offset: Int, limit: Int)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem],
+    toEntityMarshallerCompactOrganizations: ToEntityMarshaller[CompactOrganizations]
+  ): Route = {
+    val result: Future[CompactOrganizations] =
+      for {
+        _            <- validateQueryName(q)
+        pagedResults <- agreementProcessService.getAgreementProducers(q, offset = offset, limit = limit)
+      } yield CompactOrganizations(
+        results = pagedResults.results.map(t => CompactOrganization(id = t.id, name = t.name)),
+        pagination = Pagination(offset = offset, limit = limit, totalCount = pagedResults.totalCount)
+      )
+
+    onComplete(result) {
+      handleError(
+        s"Error retrieving producers from agreement filtered by eservice name $q, offset $offset, limit $limit"
+      ) orElse {
+        case Failure(_: InvalidQueryParameter) =>
+          getAgreementProducers200(
+            CompactOrganizations(results = Nil, pagination = Pagination(offset = offset, limit = limit, totalCount = 0))
+          )
+        case Success(producers)                =>
+          getAgreementProducers200(producers)
+      }
+    }
+  }
+
+  private def validateQueryName(q: Option[String]): Future[Unit] = {
```

Comment:
Just to make it less ambiguous
```suggestion
  private def validateQueryTenantName(q: Option[String]): Future[Unit] = {
```